### ISSUE TITLE
Logger reports generic cosmetic filter instead of a specific

### ISSUE THREAD

**Issue by peace2000** [uBlockOrigin/uBlock-issues#2092]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

If there's the same cosmetic filter as a generic and as a specific, the logger reports the generic being loaded instead of a specific, if the generic filter is listed before the specific filter in the filter list.

### A specific URL where the issue occurs

https://elisaviihde.fi/

### Steps to Reproduce

1. Add these to your own filters manually, in this order:
`##.ea-cookie-disclaimer`
`elisaviihde.fi##.ea-cookie-disclaimer`
2. Activate logger
3. Go to `elisaviihde.fi`

In the logger, you'll see that the generic filter is reported instead of the specific.

If you change the order:

`elisaviihde.fi##.ea-cookie-disclaimer`
`##.ea-cookie-disclaimer`

Logger will report the specific one as it should.

### Expected behavior

The logger would always report the specific filter over the generic as it was the purpose of this commit:

https://github.com/gorhill/uBlock/commit/c25938f5bcb519ee945c9ccab7d17474474dc825

### Actual behavior

The logger reports a generic cosmetic filter over a specific one, if the generic filter is listed before the specific in the list.

### uBlock Origin version

1.42.5b1

### Browser name and version

Firefox 99.0.1

### Operating System and version

Windows 10

**Comment by peace2000** [comment]
Did some digging.

1.39.2 is the last stable version where this works correctly.
1.40.0 doesn't work correctly.

**Comment by gwarser** [comment]
@peace2000 is this fixed in 1.42.4? I cannot reproduce in Nightly.

**Comment by peace2000** [comment]
No, I can repro this issue in that version as well. 1.39.2 is the last stable version where this works correctly.

@gwarser

**Comment by gwarser** [comment]
Weird. It really must be related with sequence of data in storage, because no matter what I do, I cannot reproduce in dirty profile. I can in clean installation.

**Comment by gorhill** [comment]
Hypothesis: Could be a regression from <https://github.com/gorhill/uBlock/commit/72bb89495ba0928042b8c0fe22ee57f6e955ca90>.

**Comment by peace2000** [comment]
Tested the latest dev build, seems to be working.
